src/components/Actions.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable linebreak-style */
+function getUserCountryCodeFromIp() {
+  let userCountryCode = 'CHN';
+  const url = 'https://get.geojs.io/v1/ip/country.json';
+  return fetch(url)
+    .then((res) => res.json())
+    .then((data) => {
+      userCountryCode = data.country;
+      return userCountryCode;
+    });
+}
+function getLatestDataSummary() {
+  let latestDataSummary = [];
+  const url = 'https://api.quarantine.country/api/v1/summary/latest';
+  return fetch(url)
+    .then((res) => res.json())
+    .then((res2) => {
+      latestDataSummary = res2.data;
+      return latestDataSummary;
+    });
+}
+
+function getRegions() {
+  const countrycodes = [];
+  const url = 'https://api.quarantine.country/api/v1/regions';
+  return fetch(url)
+    .then((res) => res.json())
+    .then((res2) => res2.data.forEach((element) => {
+      countrycodes[element.iso3166a2] = element.key;
+    }))
+    .then(() => {
+      countrycodes.PS = 'palestine';
+      return countrycodes;
+    });
+}
+
+function getCountryNames() {
+  const countryNames = [];
+  const url = 'https://api.quarantine.country/api/v1/regions';
+  return fetch(url)
+    .then((res) => res.json())
+    .then((res2) => res2.data.forEach((element) => {
+      countryNames[element.name] = element.key;
+    }))
+    .then(() => {
+      countryNames.Palestine = 'palestine';
+      return countryNames;
+    });
```

Comment:
I believe that you can find a way to not torture your code like this. your fetching in`getCountryNames` & `getRegions` is slightly identical, only half of a line is what differentiates them. try to make use of only one function and customize it to serve your needs